refactor(linter): remove `ModuleContentOwner` abstraction

overlookmotel opened the pull request:
Pure refactor. Remove `ModuleContentOwner` abstraction. It's just a wrapper around `AllocatorGuard` and adds no value in itself, so remove it and use `AllocatorGuard` directly.

camc314 commented:
### Merge activity

* **Jul 16, 11:02 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(linter): remove `ModuleContentOwner` abstraction (#12331)

Pure refactor. Remove `ModuleContentOwner` abstraction. It's just a wrapper around `AllocatorGuard` and adds no value in itself, so remove it and use `AllocatorGuard` directly.
